lib/data/multipleBackendGateway.js

```diff
@@ -216,6 +218,90 @@ const multipleBackendGateway = {
         });
     },
+
+    createMPU: (key, bucket, metaHeaders, websiteRedirectHeader,
+    location, log, cb) => {
+        const client = clients[location];
+        const awsKey = _createAwsKey(bucket, key, client.bucketMatch);
+        if (client.clientType === 'aws_s3') {
+            const params = { Bucket: client.awsBucketName, Key: awsKey,
+                WebsiteRedirectLocation: websiteRedirectHeader,
+                Metadata: metaHeaders };
+            return client.createMultipartUpload(params, (err, data) => {
+                if (err) {
+                    log.error('err from data backend',
+                    { error: err, dataStore: client.dataStoreName });
+                    return cb(errors.InternalError);
+                }
+                return cb(null, data);
+            });
+        }
+        const scalResponse = {};
+        scalResponse.UploadId = UUID.v4().replace(/-/g, '');
+        return cb(null, scalResponse);
+    },
+
+    uploadPart: (bucket, location, key, uploadId, partNumber, log, cb) => {
+        const client = clients[location];
+        if (client.clientType === 'aws_s3') {
+            const awsKey = _createAwsKey(bucket, key, client.bucketMatch);
+            const params = { Bucket: client.awsBucketName, Key: awsKey,
+                UploadId: uploadId, PartNumber: partNumber };
+            return client.uploadPart(params, (err, data) => {
+                if (err) {
+                    log.error('err from data backend',
+                    { error: err, dataStoreName: client.dataStoreName });
+                    return cb(errors.InternalError);
+                }
+                const dataRetrievalInfo = {
+                    key,
+                    dataStoreName: client.dataStoreName,
+                    dataStoreETag: data.ETag,
+                };
+                return cb(null, dataRetrievalInfo);
+            });
+        }
+        return cb();
+    },
+
+    completeMPU: (key, uploadId, location, jsonList, log, cb) => {
+        const client = clients[location];
+        const partList = jsonList.Part;
+        if (client.clientType === 'aws_s3') {
+            const bucket = client.awsBucketName;
+            const partArray = [];
+            return async.each(partList, (partObj, callback) => {
+                const partParams = { PartNumber: partObj.PartNumber[0],
+                ETag: partObj.ETag[0] };
+                partArray.push(partParams);
+                return callback();
+            }, () => {
+                const mpuParams = {
+                    Bucket: bucket, Key: key, UploadId: uploadId,
+                    MultipartUpload: {
+                        Parts: partArray,
+                    },
+                };
+                return client.completeMultipartUpload(mpuParams,
+                (err, data) => {
+                    if (err) {
+                        log.error('err from data backend',
```

Comment:
I would only log at error level if the error is one of the "other" errors.  For instance, if it is an InvalidPart error, this shouldn't be an error log.  more like a trace log.